### ISSUE TITLE
Infinite loop while fitting

### ISSUE THREAD

**Issue by algmyr** [vectorgraphics/asymptote#72]
While writing some simple code I ran into an issue with `asy` not terminating. It seems to be failing to fit the picture on a frame.

```
size(200);

int n = 4;
int dx = 5;
int dy = 2;

for (int i = n-1; i>=0; --i) {
    int x = i*dx;
    for (int j = -i; j <= i; j += 2) {
        int y = j*dy;
        draw((x,y)--(x+dx,y+dy), Arrow(SimpleHead));
    }
}
```
The bug is really weird and specific. For lower `n` it works fine, for `n>6` is works. For different `dx` or `dy` it works. Reversing the first loop makes it work. With any other head than `SimpleHead` (or without arrow) it works fine.

I know it calls https://github.com/vectorgraphics/asymptote/blob/f01b1baf546ed20bf49e5fd56aac4b43b4a1a5e7/base/plain_bounds.asy#L338 ad infinitum, but haven't investigated it too closely.

Edit:
It's the simplex method that gets stuck in an infinite loop in `optimize`.

**Comment by algmyr** [comment]
This exact configuration seems to somehow invalidate the assumptions made in `simplex.asy`. Specifically in `selectVar` `r.c` can become zero or even small negative numbers. Similarly the sign of `r.t[col]` can get the wrong sign.

A very dirty (and possibly incorrect in some cases) workaround is to do

    if(r.c < max*r.t[col] && r.c > 0) {

to ignore non-positive `r.c`. This however seems to be a more fundamental problem with the simplex implementation.

**Comment by johncbowman** [comment]
Thank you for your bug report. Fortunately a few months ago, a colleague and I implemented the full
simplex method in Asymptote. We will try hooking it up later this week so hopefully your bug will soon be solved.

**Comment by johncbowman** [comment]
Good news: this bug is now fixed with the robust simplex method implemented in commit 401cf60f89dcddfb6993f7bd25a86b144ec5f3ad.
